torchdynamo/test.py

```diff
@@ -0,0 +1,66 @@
+import torch
+import torch.fx
+import torch.nn.functional as F
+from typing import Callable
+import torchdynamo
+from torch.ao.quantization.quantize_fx import prepare_fx, convert_fx
+from torch.ao.quantization import default_qconfig
+
+class BasicModule(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.linear1 = torch.nn.Linear(10, 10)
+        self.scale = torch.randn(1, 10)
+
+    def forward(self, x):
+        return F.relu(self.linear1(x)) * self.scale
+
+prepare_mode = True
+class QuantizationWrapper:
+    def __init__(self, model, example):
+        super().__init__()
+        model.eval()
+        self.prepared_model = prepare_fx(model, {"": default_qconfig})
+        self.quantized_model = None
+
+    def __call__(self, *args, **kwargs):
+        if prepare_mode:
+            return self.prepared_model(*args, **kwargs)
+        else:
+            if self.quantized_model is None:
+                print("quantized model:", self.quantized_model)
+                self.quantized_model = convert_fx(self.prepared_model)
+            return self.quantized_model(*args, **kwargs)                
+            
+def quant_compiler(model, example):    
+    return QuantizationWrapper(model, example)
+    
+def quantize(m, example_inputs):
+    torchdynamo.config.debug = True
+    prepare_mode = True
```

Comment:
I think you need to add:
```suggestion
    global prepare_mode
    prepare_mode = True
```